shell/src/main/java/alluxio/cli/fs/command/LeaderCommand.java

```diff
@@ -53,6 +58,16 @@ public int run(CommandLine cl) {
       try {
         InetSocketAddress address = client.get().getAddress();
         System.out.println(address.getHostName());
+
+        List<InetSocketAddress> addresses = Arrays.asList(address);
+        MasterInquireClient inquireClient = new PollingMasterInquireClient(addresses, () ->
+                new ExponentialBackoffRetry(50, 100, 2)
+        );
+        try {
+          inquireClient.getPrimaryRpcAddress();
+        } catch (UnavailableException e) {
+          System.out.println("Failed to get the leader master RPC address.");
```

Comment:
It's better to print this to `System.err` so that if scripts parse stdout, they won't capture this warning.

Reply:
Agree, my initial code was printing to the stderr, but I ran into some issues while testing this case since the class ``AbstractAlluxioShellTest`` does not seem to provide a mechanism to capture stderr. At the moment I obtain the results of the stdout from a member variable ``mOutput``. @aaudiber Do you know if it's there is a similar mechanism to get the stderr?

Reply:
To capture stderr you could generalize `SystemOutRule` to a `RedirectStreamRule` that takes a `PrintStream` (could be either `System.out` or `System.err`) and an `OutputStream` and redirects the `PrintStream` to the `OutputStream` for the duration of the test. Then add

```java
  @Rule
  public RedirectOutputRule mErrRule = new RedirectOutputRule(System.err, mErrOutput);
```

in `AbstractAlluxioShellTest`

Reply:
Done